dags/process_elife_zip_dag.py

```diff
@@ -69,17 +79,38 @@ def extract_archived_files_to_bucket(**context):
         logger.info('ZIPPED FILES= %s', ZipFile(temp_file).namelist())
 
         # extract zip to cloud bucket
-        folder_name = zip_file_name.replace('.zip', '')
+        folder_name = zip_file_name.rstrip('.zip')
+        file_to_upload = None
+
         for zipped_file_path in ZipFile(temp_file).namelist():
-            if Path(zipped_file_path).suffix in ALLOWED_EXTENSIONS:
+
+            if is_image(zipped_file_path) and not is_allowed(zipped_file_path):
+                # convert image to jpeg
+                file_to_upload = BytesIO()
+                image = Image.open(BytesIO(ZipFile(temp_file).read(zipped_file_path)))
+                try:
+                    image.save(file_to_upload, 'JPEG')
+                except IOError:
```

Comment:
anything more specific to catch? A disk space exhausted or permission error would try a new conversion, but there is only an error on color spaces here that we are trying to work around?

Reply:
tiff images have an alpha channel that is not supported in jpeg images (as far as I understand). So instead of checking the file type (of which there could be many) this is an alternative check. In most cases PIL can convert images without having to do any conversions, meaning `image.save` should work most of the time.

Reply:
Also, these changes are made in memory.